stochastic/recorder/generate.go

```diff
@@ -0,0 +1,142 @@
+				if err != nil {
+					return nil, err
+				}
+				s.contracts.Classify(addr)
+			}
+		}
+	}
+	log.Infof("Number of storage keys for priming: %v", cfg.KeysNumber)
+	for i := int64(0); i < cfg.KeysNumber; i++ {
+		for j := i - stochastic.QueueLen - 1; j <= i; j++ {
+			if j >= 0 {
+				key, err := operations.ToHash(j)
+				if err != nil {
+					return nil, err
+				}
+				s.keys.Classify(key)
+			}
+		}
+	}
+	log.Infof("Number of storage values for priming: %v", cfg.ValuesNumber)
+	for i := int64(0); i < cfg.ValuesNumber; i++ {
+		for j := i - stochastic.QueueLen - 1; j <= i; j++ {
+			if j >= 0 {
+				value, err := operations.ToHash(j)
+				if err != nil {
+					return nil, err
+				}
+				s.values.Classify(value)
+			}
+		}
+	}
+	log.Infof("Snapshot depth: %v", cfg.KeysNumber)
+	for i := 0; i < cfg.SnapshotDepth; i++ {
+		s.snapshotFreq[i] = 1
+	}
+	for i := 0; i < operations.NumArgOps; i++ {
+		if !operations.IsValidArgOp(i) {
+			continue
+		}
+		opI, _, _, _, err := operations.DecodeArgOp(i)
+		if err != nil {
+			return nil, err
+		}
+		s.argOpFreq[i] = 1 // set frequency to greater than zero to emit operation
+		switch opI {
+		case operations.BeginSyncPeriodID:
+			j, err := operations.EncodeArgOp(operations.BeginBlockID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			s.transitFreq[i][j] = 1
+		case operations.BeginBlockID:
+			j, err := operations.EncodeArgOp(operations.BeginTransactionID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			s.transitFreq[i][j] = 1
+		case operations.EndTransactionID:
+			j1, err := operations.EncodeArgOp(operations.BeginTransactionID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			j2, err := operations.EncodeArgOp(operations.EndBlockID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			s.transitFreq[i][j1] = cfg.BlockLength - 1
+			s.transitFreq[i][j2] = 1
+		case operations.EndBlockID:
+			j1, err := operations.EncodeArgOp(operations.BeginBlockID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			j2, err := operations.EncodeArgOp(operations.EndSyncPeriodID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			s.transitFreq[i][j1] = cfg.SyncPeriodLength - 1
+			s.transitFreq[i][j2] = 1
+		case operations.EndSyncPeriodID:
+			j, err := operations.EncodeArgOp(operations.BeginSyncPeriodID, stochastic.NoArgID, stochastic.NoArgID, stochastic.NoArgID)
+			if err != nil {
+				return nil, err
+			}
+			s.transitFreq[i][j] = 1
+		default:
+			for j := 0; j < operations.NumArgOps; j++ {
+				if !operations.IsValidArgOp(j) {
+					continue
+				}
+				opJ, _, _, _, err := operations.DecodeArgOp(j)
+				if err != nil {
```

Comment:
This too

Reply:
That is right; but agian - let's leave the check for future potential errors in `DecodeArgOp()`.